spinalcordtoolbox/scripts/sct_analyze_lesion.py

```diff
@@ -553,7 +563,7 @@ def _measure_tissue_bridges(self, im_lesion_data, p_lst, idx):
 
     def _measure_length(self, im_data, p_lst, idx):
```

Comment:
I notice that the `_measure_length` method is quite short in comparison to the brand new `_measure_width` function, which computes `width` for all nonzero axial slices, then takes the max.

That made me wonder: Is the `_measure_length` logic wrong? Should we be computing `length` for all nonzero sagittal slices, then taking the max?

Instead, it seems like length is computed using any and all nonzero SI slices, regardless of whether those SI slices are all present in a single sagittal slice, which seems to be a different "maximal" computation compared to `_measure_width`.

Reply:
> I notice that the _measure_length method is quite short in comparison to the brand new _measure_width function, which computes width for all nonzero axial slices, then takes the max.

Yeah, good point; I simplified the `for` cycle below into dict comprehension in https://github.com/spinalcordtoolbox/spinalcordtoolbox/pull/4702/commits/10ea01e92fbc4c9356c70148219f129cb687297f

https://github.com/spinalcordtoolbox/spinalcordtoolbox/blob/91985398ed990465adee857b2c58eb021212673d/spinalcordtoolbox/scripts/sct_analyze_lesion.py#L582-L588